### PR TITLE
Form editing and saving inline

### PR DESCRIPTION
**Pull request by ShaunYearStrong**
picture 43
picture 44
picture 45

### CODE REVIEW

File: public/video-ui/src/components/VideoEdit/formComponents/VideoDuration.js [hunk 1]
```diff
@@ -18,14 +19,13 @@ export default class VideoDurationEdit extends React.Component {
...
-
     return (
-        <div className="form__row">
-          <label className="form__label">Duration (in seconds)</label>
-          <input { ...this.props.input} className={"form__field " + (hasError ? "form__field--error" : "")} type="number" value={this.props.video.data.duration || ""} onChange={this.onUpdateDuration} />
-          {hasError ? <p className="form__message form__message--error">{this.props.meta.error}</p> : ""}
-        </div>
+        <TextInput
+          fieldName="Duration (in seconds)"
+          fieldValue={this.props.video.data.duration}
```
Comment (currysoup) [inline public/video-ui/src/components/VideoEdit/formComponents/VideoDuration.js]: Will need to be updated with the new model changes

File: public/video-ui/src/components/VideoEdit/formComponents/VideoTitle.js [hunk 2]
```diff
@@ -18,14 +19,12 @@ export default class VideoTitleEdit extends React.Component {
...
-
     return (
-        <div className="form__row">
-          <label className="form__label">Title</label>
-          <input { ...this.props.input} className={"form__field " + (hasError ? "form__field--error" : "")} type="text" value={this.props.video.data.title || ""} onChange={this.onUpdateTitle} />
-          {hasError ? <p className="form__message form__message--error">{this.props.meta.error}</p> : ""}
-        </div>
+        <TextInput
+          fieldName="Title"
+          fieldValue={this.props.video.data.title}
```
Comment (currysoup) [inline public/video-ui/src/components/VideoEdit/formComponents/VideoTitle.js]: Will need to be updated with the new model changes

File: public/video-ui/src/components/Videos/VideoItem.js [hunk 3]
```diff
@@ -10,7 +10,7 @@ export default class VideoItem extends React.Component {
 
             <div className="grid__info">
               <div className="grid__image">
-                <img src="http://placehold.it/350x150?text=image" alt="image"/>
+                <img src={this.props.video.data.posterUrl || ''} alt={this.props.video.data.title}/>
```
Comment (currysoup) [inline public/video-ui/src/components/Videos/VideoItem.js]: Will need to be updated with the new model changes

File: public/video-ui/src/actions/VideoActions/getVideo.js [hunk 4]
```diff
@@ -29,15 +29,15 @@ export function getVideo(id) {
...
-        .catch(error => dispatch(errorReceivingVideo(error)))
-        .then(res => {
+      .then(res => {
 
-          //TODO remove this once the API returns more structured data
-          if (! res.data.metadata) {
-            res.data.metadata = {};
-          }
+        //TODO remove this once the API returns more structured data
+        if (! res.data.metadata) {
```
Comment (currysoup) [inline public/video-ui/src/actions/VideoActions/getVideo.js]: Will need to be updated with the new model changes

File: public/video-ui/src/components/Videos/VideoItem.js [hunk 5]
```diff
@@ -10,7 +10,7 @@ export default class VideoItem extends React.Component {
...
               </div>
               <p>{this.props.video.data.title}</p>
```
Comment (currysoup) [inline public/video-ui/src/components/Videos/VideoItem.js]: Will need to be updated with the new model changes

**Comment by currysoup** [comment]
Sorry about the spam. I've added comments to each line that needed a model update - github should hide them all once the patch is in which updates them.

File: public/video-ui/src/components/FormFields/FormFieldSaveWrapper.js [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+import SaveButton from '../utils/SaveButton';
+
+export default class FormFieldSaveWrapper extends React.Component {
+
+  state = {
+    editable: false
+  };
+
+  toggleEditing = () => {
+    if(this.state.editable) {
```
Comment (clloyd) [inline public/video-ui/src/components/FormFields/FormFieldSaveWrapper.js]: can this be 
```
this.setState({
  editable: !this.state.editable
});
```

File: public/video-ui/src/components/FormFields/SelectBox.js [hunk 7]
```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+
+export default class SelectBox extends React.Component {
+
+  renderField = () => {
+    if(this.props.editable) {
```
Comment (clloyd) [inline public/video-ui/src/components/FormFields/SelectBox.js]: To remove nesting can we just return within `if (!this.props.editable)` then put this code at the bottom of the function

File: public/video-ui/src/components/FormFields/TextInput.js [hunk 8]
```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+
+export default class TextInput extends React.Component {
+
+  renderField = () => {
+    if(this.props.editable) {
```
Comment (clloyd) [inline public/video-ui/src/components/FormFields/TextInput.js]: same as above

File: public/video-ui/src/components/Videos/VideoItem.js [hunk 9]
```diff
@@ -10,7 +10,7 @@ export default class VideoItem extends React.Component {
 
             <div className="grid__info">
               <div className="grid__image">
-                <img src="http://placehold.it/350x150?text=image" alt="image"/>
+                <img src={this.props.video.posterUrl || ''} alt={this.props.video.title}/>
```
Comment (clloyd) [inline public/video-ui/src/components/Videos/VideoItem.js]: Can we not use posterUrl (I don't think it exists anymore?) posterImage.master.file I think instead

File: public/video-ui/src/components/utils/SaveButton.js [hunk 10]
```diff
@@ -9,17 +9,37 @@ export default class SaveButton extends React.Component {
...
-          </button>
+        <div className="btn__group">
+          {this.saveButton()}
+          {this.resetButton()}
         </div>
     );
   };
 
+  saveButton = () => {
+    if(this.props.onSaveClick) {
```
Comment (clloyd) [inline public/video-ui/src/components/utils/SaveButton.js]: same as above

File: public/video-ui/src/components/utils/SaveButton.js [hunk 11]
```diff
@@ -9,17 +9,37 @@ export default class SaveButton extends React.Component {
...
+          <i className="i-tick-green"/>Save
+        </button>
+      )
+    } else {
+      return false;
+    }
+  };
+
+  resetButton = () => {
+    if(this.props.onResetClick) {
```
Comment (clloyd) [inline public/video-ui/src/components/utils/SaveButton.js]: Same as Above

**Pull request merged**